src/actions-util.ts

```diff
@@ -628,7 +628,15 @@ export async function sendStatusReport<S extends StatusReportBase>(
     if (isHTTPError(e)) {
       switch (e.status) {
         case 403:
-          core.setFailed(e.message || GENERIC_403_MSG);
+          if (isDependabotActor()) {
+            core.setFailed(
+              'Workflows triggered by Dependabot on the "push" event run with read-only access. ' +
+                'To use Code Scanning with Dependabot please ensure you are using the "pull_request" event for this workflow and avoid triggering on the "push" event for dependabot branches. ' +
+                "See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags for more information on how to configure these events."
```

Comment:
Minor tweaks, but in general I think this is good.
Should we link instead to https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-pull-requests? (That in turn links to the URL you have here.)

```suggestion
              'Workflows triggered by Dependabot on the "push" event run with read-only access. ' +
                'To use Code Scanning with Dependabot, please ensure you are using the "pull_request" event for this workflow and avoid triggering on the "push" event for Dependabot branches. ' +
                "See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags for more information on how to configure these events."
```

Reply:
Hmm, I agree https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-pull-requests could be good. Would it better to link to the `scanning-on-push` section which is immediately above? That way I expect a user would read that and the one below, and there's a big warning box about scanning pull requests. If we link to the pull requests section I worry stuff above that might be missed.